libraries/aws/provisioning/test/unit/aws_iot_tests_provisioning_parser.c

```diff
@@ -484,6 +600,190 @@ TEST( Provisioning_Unit_Parser, TestParseKeysAndCertificateResponseWithMissingEn
+        0x6D,                                                                               /* # text(13) */
+        0x63, 0x65, 0x72, 0x74, 0x69, 0x66, 0x69, 0x63, 0x61, 0x74, 0x65, 0x49, 0x64,       /* # "certificateId" */
+        0x0A,                                                                               /* # unsigned(10) */
+        0x78, 0x19,                                                                         /*# text(25) */
+        0x63, 0x65, 0x72, 0x74, 0x69, 0x66, 0x69, 0x63, 0x61, 0x74, 0x65, 0x4F, 0x77, 0x6E,
+        0x65, 0x72, 0x73, 0x68, 0x69, 0x70, 0x54, 0x6F, 0x6B, 0x65, 0x6E,                   /*# "certificateOwnershipToken"
+                                                                                             * */
+        0x66,                                                                               /*# text(6) */
+        0x54, 0x6F, 0x6B, 0x65, 0x6E, 0x21                                                  /*# "Token!" */
+    };
+
+
+
+    TEST_ASSERT_EQUAL( AWS_IOT_PROVISIONING_BAD_RESPONSE, _AwsIotProvisioning_ParseCsrResponse( AWS_IOT_ACCEPTED,
+                                                                                                payloadWithInvalidCertId,
+                                                                                                sizeof( payloadWithInvalidCertId ),
+                                                                                                &wrapperCallback ) );
+
+    /*************** Response payload with invalid ownership token data********************/
+    const uint8_t payloadWithInvalidToken[] =
+    {
+        0xA3,                                                                               /* # map( 3 ) */
+        0x6E,                                                                               /* # text( 14 ) */
+        0x63, 0x65, 0x72, 0x74, 0x69, 0x66, 0x69, 0x63, 0x61, 0x74, 0x65, 0x50, 0x65, 0x6D, /* # "certificatePem" */
+        0x67,                                                                               /* # text(7) */
+        0x61, 0x62, 0x63, 0x64, 0x65, 0x66, 0x67,                                           /* # "abcdefg" */
+        0x6D,                                                                               /* # text(13) */
+        0x63, 0x65, 0x72, 0x74, 0x69, 0x66, 0x69, 0x63, 0x61, 0x74, 0x65, 0x49, 0x64,       /* # "certificateId" */
+        0x66,                                                                               /* # text(6) */
+        0x68, 0x69, 0x6A, 0x6B, 0x6C, 0x6D,                                                 /* # "hijklm" */
+        0x78, 0x19,                                                                         /*# text(25) */
+        0x63, 0x65, 0x72, 0x74, 0x69, 0x66, 0x69, 0x63, 0x61, 0x74, 0x65, 0x4F, 0x77, 0x6E,
+        0x65, 0x72, 0x73, 0x68, 0x69, 0x70, 0x54, 0x6F, 0x6B, 0x65, 0x6E,                   /*# "certificateOwnershipToken"
```

Comment:
Nit: alignment for the `#` with the other rows

Reply:
Oops. Missed the comment. Resolved in https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/836